src/metabase/api/collection.clj

```diff
@@ -425,12 +426,7 @@
                                              [:= :mr.moderated_item_type (h2x/literal "card")]]
                    [:core_user :u] [:= :u.id :r.user_id]]
        :where     [:and
-                   (collection/visible-collection-filter-clause :collection_id
-                                                                {:include-archived-items :all
-                                                                 :permission-level (if archived?
-                                                                                     :write
-                                                                                     :read)
-                                                                 :archive-operation-id nil})
+                   (collection/visible-collection-filter-clause :collection_id {:cte-name :visible_collection_ids})
```

Comment:
To clarify what's happening here: I'm passing the the name of the CTE into `visible-collection-filter-clause` for it to use instead of doing a subquery. The CTE is added to the top-level of the query below. 

It's a bit indirect but I wasn't sure of a better way to approach this without fully rearchitecting this code